api/pkg/apis/v1alpha1/managers/stage/stage-manager.go

```diff
@@ -738,19 +750,15 @@ func (s *StageManager) HandleTriggerEvent(ctx context.Context, campaign model.Ca
 					return status, activationData
 				}
 			}
+			// sVal is empty, no next stage
 			status.NextStage = sVal
-			if pauseRequested {
-				status.IsActive = false
-				status.Status = v1alpha2.Paused
-				status.StatusMessage = v1alpha2.Paused.String()
-			} else {
-				status.IsActive = false
-				status.Status = v1alpha2.Done
-				status.StatusMessage = v1alpha2.Done.String()
-			}
+			status.IsActive = false
+			status.Status = v1alpha2.Done
```

Comment:
If a stage is scheduled it shouldn't be marked as Done immediately, right?

Reply:
In line 665 - 691, we already check pauseRequested which is true when the stage is scheduled. Instead, we should check pauseRequested when selfDriving is false as well (At line 761)?